src/abacus-ts/calculators/blockRewards.ts

```diff
@@ -0,0 +1,21 @@
+import { IndexerHistoricalBlockTradingReward } from '@/types/indexer/indexerApiGen';
+import { keyBy, maxBy } from 'lodash';
+
+import { MustBigNumber } from '@/lib/numbers';
+
+import { Loadable } from '../lib/loadable';
+import { mapLoadableData } from '../lib/mapLoadable';
+import { mergeObjects } from '../lib/mergeObjects';
+
+export function calculateBlockRewards(
+  liveTransfers: Loadable<IndexerHistoricalBlockTradingReward[]>,
+  restTransfers: Loadable<IndexerHistoricalBlockTradingReward[]>
+) {
+  const getRewardsById = (data: Loadable<IndexerHistoricalBlockTradingReward[]>) =>
+    mapLoadableData(data, (d) => keyBy(d, (reward) => reward.createdAtHeight));
+  return mergeObjects(
+    getRewardsById(liveTransfers).data ?? {},
+    getRewardsById(restTransfers).data ?? {},
+    (first, second) => maxBy([first, second], (f) => MustBigNumber(f.createdAtHeight).toNumber())!
```

Comment:
does createdAtHeight need to be casted to BigNumber? would ParseInt or Number(...) not work?

Reply:
Yeah those would work to, I just for some reason trust the bignumber parser more.

Reply:
kk. just making sure because I think that createdAtHeight will always be an int and MustBigNumber might be overkill